identity/verifier_test.go

```diff
@@ -5,26 +5,86 @@ import (
 	"testing"
 )
 
-func TestVerifier_VerifyWhenSignatureIsCorrect(t *testing.T) {
+func TestVerifier_extractSignerWhenSignatureIsCorrect(t *testing.T) {
 	message := []byte("Boop!")
 	signature := SignatureHex("1f89542f406b2d638fe09cd9912d0b8c0b5ebb4aef67d52ab046973e34fb430a1953576cd19d140eddb099aea34b2985fbd99e716d3b2f96a964141fdb84b32000")
 
-	verifier := NewVerifier(FromAddress("0x53a835143c0ef3bbcbfa796d7eb738ca7dd28f68"))
+	signerId, err := extractSignerIdentity(message, signature)
+	assert.NoError(t, err)
+	assert.Exactly(t, FromAddress("0x53a835143c0ef3bbcbfa796d7eb738ca7dd28f68"), signerId)
+}
+
+func TestVerifier_extractSignerWhenSignatureIsEmpty(t *testing.T) {
+	message := []byte("boop!")
+	signature := SignatureHex("")
+
+	signerId, err := extractSignerIdentity(message, signature)
+	assert.EqualError(t, err, "Signature is empty")
+	assert.Exactly(t, Identity{}, signerId)
+}
+
+func TestVerifier_extractSignerWhenSignatureIsMalformed(t *testing.T) {
+	message := []byte("boop!")
+	signature := SignatureHex("7369676e6564")
+
+	signerId, err := extractSignerIdentity(message, signature)
+	assert.EqualError(t, err, "invalid signature length")
+	assert.Exactly(t, Identity{}, signerId)
+}
+
+func TestVerifier_extractSignerWhenMessageIsChanged(t *testing.T) {
```

Comment:
I think "IsChanged" in the name does not reflect the actual test. I would expect an error test here.

Reply:
That's exactly what I checkin g - if attacker changes a payload body